interfaces/api/service/IFrontEnd.h

```diff
@@ -390,6 +391,44 @@ class XPCF_CLIENTUUID("91a569da-5695-11ec-bf63-0242ac130002") XPCF_SERVERUUID("9
                                             const std::string & mapUUID,
                                             SRef<SolAR::datastructure::PointCloud> & pointCloud) const = 0;
 
+    /// @brief Request for a map processing giving the type of process to apply (asynchronous)
+    /// @param[in] keycloakToken a valid Keycloak Token collected by client after login to the Keycloak server
+    /// @param[in] mapUUID the UUID of the map to process
+    /// @param[in] processingType the type of process to apply on the map
+    /// @return
+    /// * FrameworkReturnCode::_SUCCESS if processing is able to proceed
+    /// * FrameworkReturnCode::_NO_SERVICE_AVAILABLE if a necessary service is not available
+    /// * else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode requestForMapProcessing(const std::string & keycloakToken,
+                                                        const std::string & mapUUID,
+                                                        const MapProcessingType & processingType) = 0;
+
+    /// @brief Get status and progress percentage concerning a map processing in progress
+    ///        If status = COMPLETED then give the map UUID of the new resulting map
+    /// @param[in] keycloakToken a valid Keycloak Token collected by client after login to the Keycloak server
+    /// @param[in] mapUUID the UUID of the map being processed
+    /// @param[out] status the current map processing status
+    /// @param[out] progress the current progress percentage (valid value should be between 0 and 1)
+    /// @param[out] resultingMapUUID the map UUID of the new created map (processing result)
+    /// @return FrameworkReturnCode::_SUCCESS if the status and progress are available, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode getStatusForMapProcessing(const std::string & keycloakToken,
+                                                          const std::string & mapUUID,
+                                                          MapProcessingStatus status,
```

Comment:
Here `status` should not be passed by value since it's an 'out' parameter, right?

Reply:
done